databricks_cli/configure/config.py

```diff
@@ -90,57 +96,52 @@ def overwrite(self):
             self._config.write(cfg)
         os.chmod(config_path, 0o600)
 
-    @property
-    def is_valid(self):
-        return self.is_valid_with_password or self.is_valid_with_token
+    def is_valid(self, profile):
+        return self.is_valid_with_password(profile) or self.is_valid_with_token(profile)
 
-    @property
-    def is_valid_with_password(self):
-        return self._config.has_option(DEFAULT_SECTION, USERNAME) and \
-            self._config.has_option(DEFAULT_SECTION, PASSWORD) and \
-            self._config.has_option(DEFAULT_SECTION, HOST)
+    def is_valid_with_password(self, profile):
+        return self._config.has_option(profile, USERNAME) and \
+            self._config.has_option(profile, PASSWORD) and \
+            self._config.has_option(profile, HOST)
 
-    @property
-    def is_valid_with_token(self):
-        return self._config.has_option(DEFAULT_SECTION, TOKEN) and \
-            self._config.has_option(DEFAULT_SECTION, HOST)
+    def is_valid_with_token(self, profile):
+        return self._config.has_option(profile, TOKEN) and \
+            self._config.has_option(profile, HOST)
 
-    @property
-    def host(self):
-        return self._config.get(DEFAULT_SECTION, HOST) if self.is_valid else None
+    def host(self, profile):
```

Comment:
Seems more extensible if we implement a `DatabricksConfigProvider` class (or `Factory`) that has `def from_profile(profile):` method. Then the code changes to

~~~python
conf = DatabricksConfProvider.from_profile(profile)
api_client = ApiClient(conf)
~~~